File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/BuildInfoCollector.java
package cn.tongdun.mobrisk.core.collectors;

import android.os.Build;

/**
 * @description: BuildInfo
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class BuildInfoCollector {

    public String getModel() {
        try {
            return Build.MODEL;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getManufacturer() {
        try {
            return Build.MANUFACTURER;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getAndroidVersion() {
        try {
            return Build.VERSION.RELEASE;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getSdkVersion() {
        try {
            return String.valueOf(Build.VERSION.SDK_INT);
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getKernelVersion() {
        try {
            return System.getProperty("os.version");
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getFingerprint() {
        try {
            return Build.FINGERPRINT;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getHardware() {
        try {
            return Build.HARDWARE;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getBrand() {
        try {
            return Build.BRAND;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getHost() {
        try {
            return Build.HOST;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getDisplay() {
        try {
            return Build.DISPLAY;
        } catch (Exception ignored) {
        }
        return "";
    }

    public String getProduct() {
        try {
            return Build.PRODUCT;
        } catch (Exception ignored) {
        }
        return "";
    }

}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/BuildInfoCollector.kt
package cn.tongdun.mobrisk.core.collectors

import android.os.Build
import cn.tongdun.mobrisk.core.tools.executeSafe

/**
 * @description: Build Info
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface BuildInfoInterface {
    fun getModel(): String
    fun getManufacturer(): String
    fun getAndroidVersion(): String
    fun getSdkVersion(): String
    fun getKernelVersion(): String
    fun getFingerprint(): String
    fun getHardware(): String
    fun getBrand(): String
    fun getHost(): String
    fun getDisplay(): String
    fun getProduct(): String
}

class BuildInfoCollector : BuildInfoInterface {
    override fun getModel(): String = executeSafe({ Build.MODEL }, "")

    override fun getManufacturer(): String = executeSafe({ Build.MANUFACTURER }, "")

    override fun getAndroidVersion(): String = executeSafe({ Build.VERSION.RELEASE }, "")

    override fun getSdkVersion(): String = executeSafe({ Build.VERSION.SDK_INT.toString() }, "")

    override fun getKernelVersion(): String =
        executeSafe({ System.getProperty("os.version") ?: "" }, "")

    override fun getFingerprint(): String = executeSafe({ Build.FINGERPRINT }, "")

    override fun getHardware(): String = executeSafe({ Build.HARDWARE }, "")

    override fun getBrand(): String = executeSafe({ Build.BRAND }, "")

    override fun getHost(): String = executeSafe({ Build.HOST }, "")

    override fun getDisplay(): String = executeSafe({ Build.DISPLAY }, "")

    override fun getProduct(): String = executeSafe({ Build.PRODUCT }, "")
}